spec/per_seat_server_spec.rb

```diff
@@ -0,0 +1,145 @@
+      customerId: customer_id,
+      paymentMethodId: "pm_card_chargeCustomerFail",
+      priceId: "BASIC",
+      quantity: 3,
+    })
+    expect(status).to eq(200)
+    expect(incomplete_subscription).not_to be_nil
+    expect(incomplete_subscription["status"]).to eq("incomplete")
+    expect(incomplete_subscription["latest_invoice"]["id"]).to start_with("in_")
+    invoice_id = incomplete_subscription["latest_invoice"]["id"]
+
+    # Create the subscription with a bad price
+    invoice, status = post_json("/retry-invoice", {
+      paymentMethodId: "pm_card_visa",
+      customerId: customer_id,
+      invoiceId: invoice_id,
+    })
+    expect(status).to eq(200)
+    expect(invoice).not_to be_nil
+    expect(invoice["id"]).to start_with("in_")
+
+    # Get sub info
+    sub_info, _ = post_json("/retrieve-subscription-information", {
+      subscriptionId: subscription_id,
+    })
+    expect(sub_info.keys).to contain_exactly(
+      "card",
+      "product_description",
+      "current_price",
+      "current_quantity",
+      "latest_invoice",
+      "upcoming_invoice"
+    )
+
+    # Get upcoming invoice with new price
+    upcoming_invoice, _ = post_json("/retrieve-upcoming-invoice", {
+      customerId: customer_id,
+      newPriceId: "PREMIUM",
+      quantity: 3,
+      subscriptionId: subscription_id
+    })
+    expect(upcoming_invoice).not_to be_nil
+    expect(upcoming_invoice.keys).to contain_exactly("immediate_total", "next_invoice_sum", "invoice")
+
+    # Get upcoming invoice with new quantity
+    upcoming_invoice, _ = post_json("/retrieve-upcoming-invoice", {
+      customerId: customer_id,
+      newPriceId: "BASIC",
+      quantity: 5,
+      subscriptionId: subscription_id
+    })
+    expect(upcoming_invoice).not_to be_nil
+    expect(upcoming_invoice.keys).to contain_exactly("immediate_total", "next_invoice_sum", "invoice")
+
+    # Get upcoming invoice with out subscription
+    upcoming_invoice, _ = post_json("/retrieve-upcoming-invoice", {
+      customerId: customer_id,
+      newPriceId: "BASIC",
+      quantity: 5,
+    })
+    expect(upcoming_invoice).not_to be_nil
+    expect(upcoming_invoice.keys).to contain_exactly("invoice")
+
+    # Update subscription with new quantity
+    updated_subscription, _ = post_json("/update-subscription", {
+      customerId: customer_id,
+      newPriceId: "BASIC",
+      quantity: 5,
+      subscriptionId: subscription_id
+    })
+    expect(updated_subscription).not_to be_nil
+    expect(updated_subscription.keys).to contain_exactly("subscription")
+
+    # Update subscription with new price
+    updated_subscription, _ = post_json("/update-subscription", {
+      customerId: customer_id,
+      newPriceId: "PREMIUM",
+      quantity: 5,
+      subscriptionId: subscription_id
+    })
+    expect(updated_subscription).not_to be_nil
+    expect(updated_subscription.keys).to contain_exactly("subscription")
+
```

Comment:
confirm quantity was updated in the call?

Reply:
good one. added above when the update should succeed with the quantity change.